htpasswd.go

```diff
@@ -177,30 +216,66 @@ func (h *HtpasswdFile) ReloadCheck() error {
 func (h *HtpasswdFile) Validate(user string, password string) bool {
 	_ = h.ReloadCheck()
 
+	hash := sha256.New()
+	// hash.Write can never fail
+	_, _ = hash.Write([]byte(user))
+	_, _ = hash.Write([]byte(":"))
+	_, _ = hash.Write([]byte(password))
+
 	h.mutex.Lock()
-	realPassword, exists := h.Users[user]
+	// avoid race conditions with cache replacements
```

Comment:
Again, this mutex isn't needed if h.users and h.cache are sync.Map.

Reply:
h.users and h.cache should be replaced atomically. sync.Map can only provide synchronization within a map but not across maps.